packages/vue-apollo-composable/src/useApolloClient.ts

```diff
@@ -8,11 +8,11 @@ type ClientId = string
 type ClientDict<T> = Record<ClientId, ApolloClient<T>>
 
 export interface UseApolloClientReturn<TCacheShape> {
-  resolveClient: (clientId?: ClientId) => ApolloClient<TCacheShape>
-  readonly client: ApolloClient<TCacheShape>
+  resolveClient: (clientId?: ClientId) => ApolloClient<TCacheShape> | null
```

Comment:
Why the `null`s here?

Reply:
`useApolloClient` returns the variable `currentApolloClient` of type `ApolloClient<TCacheShape> | null` without performing any checks. This variable may be uninitialized or null.

```ts
export function useApolloClient<TCacheShape = any>(clientId?: ClientId): UseApolloClientReturn<TCacheShape> {
  if (!getCurrentInstance()) {
    return {
      resolveClient: () => currentApolloClient,
      get client () { return currentApolloClient }
    }
  }
```

However, this would be easy to fix if `useApolloClient` used an assertion like `resolveDefaultClient` and `resolveClientWithId`.

```ts
export function useApolloClient<TCacheShape = any>(clientId?: ClientId): UseApolloClientReturn<TCacheShape> {
  if (!getCurrentInstance()) {
    if (!currentApolloClient) {
      throw new Error('Apollo Client could not be found')
    }

    return {
      resolveClient: () => currentApolloClient,
      get client () { return currentApolloClient }
    }
  }
```